provider/go.mod

```diff
@@ -2,15 +2,13 @@ module github.com/pulumi/pulumi-eks/provider/v3
 
 go 1.23
 
-toolchain go1.23.4
```

Comment:
Why remove?  I see it is being added to the sdk/go.mod.

Reply:
This PR is combining all dependabot PRs, and this is being removed in: https://github.com/pulumi/pulumi-eks/pull/1702/files